src/components/ErrorComponents/errorParser.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * This hook will take an error caught from a promise,
+ * and attempt to parse it and retrieve any message, code, and stacktrace
+ */
+
+// Passing in the intl object isn't ideal,
+// but this can't be a hook if it's going to run conditionally
+const errorParser = async (error, intl) => {
+  let errorMessage;
+  let errorBody;
+
+  /* For HTTPError we assume error.response returns a Promise */
+  if (error.name.toLowerCase() === 'httperror') {
+    /*
+     * Due to the nature of okapi endpoint calls, err.response
+     * could comprise of JSON data OR string data. Read in as Blob
+     * and deal with either case separately
+     */
+    const errorBlob = await error.response?.blob();
```

Comment:
Lets avoid accessing properties of error throughout this file and destructure them at the top with default values